stdlib/_pickle.pyi

```diff
@@ -0,0 +1,104 @@
+import sys
+from _typeshed import ReadableBuffer, SupportsWrite
+from collections.abc import Callable, Iterable, Iterator, Mapping
+from pickle import PickleBuffer as PickleBuffer
+from typing import Any, Protocol
+from typing_extensions import TypeAlias
+
+class _ReadableFileobj(Protocol):
+    def read(self, n: int, /) -> bytes: ...
+    def readline(self) -> bytes: ...
+
+# This protocol approximates the undocumented PicklerMemoProxy
+# and UnpicklerMemoProxy types.
+class _MemoProxy(Protocol):
+    def clear(self, /) -> None: ...
+    def copy(self, /) -> dict[int, tuple[int, Any]]: ...
```

Comment:
I'm of two minds here. I like the protocol for the getter return type; it avoids having the PicklerMemoProxy class in the stubs, where it could imply it's importable, and it isn't. I might feel better about that if it was named `_PicklerMemoProxy`, but alas.

I dislike the protocol for the setter value, because that method actually does do an identity check on PicklerMemoProxy or UnpicklerMemoProxy, depending.

I have been pondering this since I opened the MR, and have flipped back and forth which I think is better a couple times now.